### PR TITLE
TEST 89 As a user I should be able to see list of all users

### PR DESCRIPTION
**Pull request by kristiyanplpeev**
Backend - added new module users with endpoint for getting all users.

Frontend - added navigation at the top bar, admin home view, and view with all listed users.

### CODE REVIEW

File: client/src/components/SideBar/SideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
+import "./SideBar.css";
+import { RouteComponentProps, withRouter } from "react-router";
+
+interface SidebarProps extends RouteComponentProps {}
+
+class SideBar extends Component<SidebarProps> {
+  render(): JSX.Element {
+    if (!this.props.location.pathname.includes("/admin")) return <></>;
```
Comment (MilenIvanovv) [inline client/src/components/SideBar/SideBar.tsx:16]: I think App.tsx is more suitable for this logic. Everything about url/paths is in there.

File: client/src/components/AdminUsers/AdminUsers.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+interface AdminUsersProps {}
+
+interface AdminUserState {
+  error: boolean;
+  loading: boolean;
+  users: Array<IUser>;
+}
+
+class AdminUsers extends Component<AdminUsersProps, AdminUserState> {
```
Comment (MilenIvanovv) [inline client/src/components/AdminUsers/AdminUsers.tsx:26]: I think just Users or UsersList is a more suitable name for this component. I thought that AdminUsers renders only users that are with role admin.

File: server/src/users/users.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [UsersController],
+    }).compile();
+
+    controller = module.get<UsersController>(UsersController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
```
Comment (MilenIvanovv) [inline server/src/users/users.controller.spec.ts:16]: What are the benefits of this test?

File: server/src/users/users.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [UsersService],
+    }).compile();
+
+    service = module.get<UsersService>(UsersService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
```
Comment (MilenIvanovv) [inline server/src/users/users.service.spec.ts:16]: What are the benefits of this test?

**Review by MilenIvanovv — CHANGES_REQUESTED** [review]
LGTM, left some comments with a question and for minor changes

File: client/src/components/SideBar/SideBar.tsx (hunk 1 above)
Comment (kristiyanplpeev) [reply client/src/components/SideBar/SideBar.tsx:16]: done.

File: client/src/components/AdminUsers/AdminUsers.tsx (hunk 2 above)
Comment (kristiyanplpeev) [reply client/src/components/AdminUsers/AdminUsers.tsx:26]: done.

File: server/src/users/users.controller.spec.ts (hunk 3 above)
Comment (kristiyanplpeev) [reply server/src/users/users.controller.spec.ts:16]: done.

File: server/src/users/users.service.spec.ts (hunk 4 above)
Comment (kristiyanplpeev) [reply server/src/users/users.service.spec.ts:16]: done.

**Review by MilenIvanovv — APPROVED** [review]
LGTM

**Pull request merged**